setup/env.sh

```diff
@@ -172,11 +205,13 @@ function model_attribute {
   esac
 
   local modelcomponents=$(echo $model | cut -d '/' -f 2 |  tr '[:upper:]' '[:lower:]' | $LLMDBENCH_CONTROL_SCMD -e 's^qwen^qwen-^g' -e 's^-^\n^g')
+  local provider=$(echo $model | cut -d '/' -f 1)
   local type=$(echo "${modelcomponents}" | grep -Ei "nstruct|hf|chat|speech|vision")
-  local parameters=$(echo "${modelcomponents}" | grep -Ei "[0-9].*b" | $LLMDBENCH_CONTROL_SCMD -e 's^a^^' -e 's^\.^p^')
-  local majorversion=$(echo "${modelcomponents}" | grep -Ei "^[0-9]" | grep -Evi "b|E" | cut -d '.' -f 1)
+  local parameters=$(echo "${modelcomponents}" | grep -Ei "[0-9].*b|[0-9].*m" | $LLMDBENCH_CONTROL_SCMD -e 's^a^^' -e 's^\.^p^')
+  local majorversion=$(echo "${modelcomponents}" | grep -Ei "^[0-9]" | grep -Evi "b|E" |  $LLMDBENCH_CONTROL_SCMD -e "s/$parameters//g" | cut -d '.' -f 1)
   local kind=$(echo "${modelcomponents}" | head -n 1 | cut -d '/' -f 1)
-  local label=${kind}-${majorversion}-${parameters}
+  local label=$(echo ${kind}-${majorversion}-${parameters} | $LLMDBENCH_CONTROL_SCMD -e 's^-$^^g' -e 's^--^^g')
+  local folder=$(echo $model | $LLMDBENCH_CONTROL_SCMD -e 's^/^_^g' -e 's^-^_^g')
```

Comment:
if this was created for the modelservice work; the PR for that does something slightly different.

Reply:
You're right. I can leave this "as is" and we can incorporate your `as_label` too